### PR TITLE
Deprecate updating admin credentials using API calls

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
Deprecate updating admin credentials using API calls

## Motivation and Context
Root credentials are not allowed to change in all of our
distributed setup deployments, this PR simply removes
that behavior.

## Regression
No

## How Has This Been Tested?
Well this only removes the functionality.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :x: [more...](#761b2289-4dd8-4725-a6c4-1a0b14441a17) |
| mint-compression-fs.sh | :x: [more...](#d30b0fc1-eab8-43bf-8732-eed09772f253) |
| mint-dist-xl.sh | :x: [more...](#45de209b-f5d1-42b5-b3ee-e5763a3cbadf) |
<a name="45de209b-f5d1-42b5-b3ee-e5763a3cbadf"></a>
### 7570-01ce17b/mint-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.56:30811
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 563b0509acc2:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 0 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 45 seconds
(4/14) Running aws-sdk-ruby tests ... done in 3 seconds
(5/14) Running awscli tests ... done in 1 minutes and 51 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 27 seconds
(8/14) Running minio-dotnet tests ... done in 36 seconds
(9/14) Running minio-go tests ... done in 1 minutes and 6 seconds
(10/14) Running minio-java tests ... done in 2 minutes and 17 seconds
(11/14) Running minio-js tests ... done in 42 seconds
(12/14) Running minio-py tests ... FAILED in 4 minutes and 10 seconds
{
  "status": "FAIL",
  "name": "minio-py:test_presigned_put_object_expiry_5sec",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1347, in test_presigned_put_object_expiry_5sec\n    raise ValueError('Presigned put url failed to expire!')\nValueError: Presigned put url failed to expire!\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1894, in main\n    test_presigned_put_object_expiry_5sec(client, log_output)\n  File \"/mint/run/core/minio-py/tests.py\", line 1349, in test_presigned_put_object_expiry_5sec\n    raise Exception(err)\nException: Presigned put url failed to expire!\n",
  "args": {
    "bucket_name": "minio-py-test-f9265463-6a2d-49b4-b743-e68ce6e598d1",
    "object_name": "65920cbe-34cf-421a-a501-c2b63b36ae6a"
  },
  "function": "presigned_put_object(bucket_name, object_name, expires)",
  "message": "Presigned put url failed to expire!",
  "duration": 5148
}

Executed 11 out of 14 tests successfully.
```
<a name="d30b0fc1-eab8-43bf-8732-eed09772f253"></a>
### 7570-01ce17b/mint-compression-fs.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.61:32746
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp cb7b6928c5e6:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 0 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 42 seconds
(4/14) Running aws-sdk-ruby tests ... done in 2 seconds
(5/14) Running awscli tests ... done in 1 minutes and 45 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 18 seconds
(8/14) Running minio-dotnet tests ... done in 23 seconds
(9/14) Running minio-go tests ... done in 26 seconds
(10/14) Running minio-java tests ... done in 1 minutes and 0 seconds
(11/14) Running minio-js tests ... done in 36 seconds
(12/14) Running minio-py tests ... FAILED in 2 minutes and 21 seconds
{
  "duration": 57,
  "args": {
    "bucket_name": "minio-py-test-4f0b325e-44fa-479c-87da-b21a241f63de",
    "object_name": "2d4ce207-9891-44af-b45a-63c6f126b464"
  },
  "message": "AccessDenied: message: Access Denied",
  "name": "minio-py:test_presigned_get_object_expiry_5sec",
  "function": "presigned_get_object(bucket_name, object_name, expires, response_headers, request_date)",
  "status": "FAIL",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1220, in test_presigned_get_object_expiry_5sec\n    object_name).get_exception()\nminio.error.AccessDenied: AccessDenied: message: Access Denied\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1885, in main\n    test_presigned_get_object_expiry_5sec(client, log_output)\n  File \"/mint/run/core/minio-py/tests.py\", line 1228, in test_presigned_get_object_expiry_5sec\n    raise Exception(err)\nException: AccessDenied: message: Access Denied\n"
}

Executed 11 out of 14 tests successfully.
```
<a name="761b2289-4dd8-4725-a6c4-1a0b14441a17"></a>
### 7570-01ce17b/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.57:31920
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp e1ac43b65eea:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... FAILED in 3 seconds
{
  "alert": "",
  "args": {
    "bucketName": "aws-sdk-go-test-59foz62vfnr9ux",
    "expiry": 60000000000,
    "objectName": "presignedTest"
  },
  "duration": 3062,
  "error": "RequestError: send request failed\ncaused by: Put http://72.28.97.57:31920/aws-sdk-go-test-59foz62vfnr9ux: dial tcp 72.28.97.57:31920: connect: no route to host",
  "function": "PresignedPut",
  "message": "AWS SDK Go CreateBucket Failed",
  "name": "aws-sdk-go",
  "status": "FAIL"
}

Executed 0 out of 14 tests successfully.
```

**Review by kannappanr — DISMISSED** [review]
LGTM

**Comment by kannappanr** [comment]
`cmd/admin-handlers_test.go:638:29: undefined: madmin.SetCredsReq`

Travis Errors needs to be fixed.

File: pkg/madmin/README.md [hunk 1]
```diff
@@ -1,120 +1,539 @@
...
+| [`ServiceStatus`](#ServiceStatus)         | [`ServerInfo`](#ServerInfo)                 | [`Heal`](#Heal)    | [`GetConfig`](#GetConfig)         | [`TopLocks`](#TopLocks) | [`AddUser`](#AddUser)                 |                                                   |
+| [`ServiceSendAction`](#ServiceSendAction) | [`ServerCPULoadInfo`](#ServerCPULoadInfo)   |                    | [`SetConfig`](#SetConfig)         |                         | [`SetUserPolicy`](#SetUserPolicy)     | [`StartProfiling`](#StartProfiling)               |
+|                                           | [`ServerMemUsageInfo`](#ServerMemUsageInfo) |                    | [`GetConfigKeys`](#GetConfigKeys) |                         | [`ListUsers`](#ListUsers)             | [`DownloadProfilingData`](#DownloadProfilingData) |
+|                                           |                                             |                    | [`SetConfigKeys`](#SetConfigKeys) |                         | [`AddCannedPolicy`](#AddCannedPolicy) |                                                   |
 
-You need four items to connect to MinIO admin services.
 
+## 1. Constructor
+<a name="MinIO"></a>
```
Comment (ebozduman) [inline pkg/madmin/README.md:53]: It is just a minor point, but this anchor, `<a name="MinIO"></a>`, is useless.

**Review by ebozduman — CHANGES_REQUESTED** [review]

**Review by ebozduman — APPROVED** [review]
LGTM

**Review by kannappanr — APPROVED** [review]

**Pull request merged**